Mlem/API/Internal/SiteVersion.swift

```diff
@@ -0,0 +1,200 @@
+//
+//  APISiteVersionNumber.swift
+//  Mlem
+//
+//  Created by Sjmarf on 09/09/2023.
+//
+import Foundation
+
+enum SiteVersionSuffix: Equatable {
+    /// The only type other than beta "-beta.x" and releaseCandidate "-rc.x" that I've seen
+    /// is lemmy.blahaj.zone's "-kt.x". I'm unable to determine the meaning of this.
+    case beta(Int), releaseCandidate(Int), other(String, Int? = nil)
+}
+
+extension SiteVersionSuffix: CustomStringConvertible {
+    var description: String {
+        switch self {
+        case .beta(let int):
+            return "beta.\(int)"
+        case .releaseCandidate(let int):
+            return "rc.\(int)"
+        case .other(let string, let int):
+            if let int = int {
+                return "\(string).\(int)"
+            }
+            return string
+        }
+    }
+}
+
+extension SiteVersionSuffix: Comparable {
+    static func < (lhs: SiteVersionSuffix, rhs: SiteVersionSuffix) -> Bool {
+        switch (lhs, rhs) {
+        case (.beta, .releaseCandidate):
+            return true
+        case (.releaseCandidate, .beta):
+            return false
+        case (.releaseCandidate(let iteration1), .releaseCandidate(let iteration2)):
+            return iteration1 < iteration2
+        case (.beta(let iteration1), .beta(let iteration2)):
+            return iteration1 < iteration2
+        case (.other(let string1, let iteration1), .other(let string2, let iteration2)):
+            guard string1 == string2 else {
+                return false
+            }
+            if let iteration1 = iteration1, let iteration2 = iteration2 {
+                return iteration1 < iteration2
+            }
+            return false
+        case (.beta, .other), (.releaseCandidate, .other):
+            return true
+        default:
+            return false
+        }
+    }
+}
+
+enum SiteVersion: Equatable {
+    
+    /// Format 0.18.2
+    case release(major: Int, minor: Int, patch: Int)
+    /// Format 0.18.2-beta.5, 0.18.2-rc.2
+    case suffixed(major: Int, minor: Int, patch: Int, suffix: SiteVersionSuffix)
+    case other(String)
+    case zero
+    case infinity
```

Comment:
That is correct. They're designed to be used outside of the SiteVersion in situations such as this 

```swift 
    var minimumVersion: APISiteVersion {
        switch self {
        case .topThreeMonth, .topSixMonth, .topNineMonth:
            return SiteVersion("0.18.1")
        default:
            return .zero
        }
```

Of course, there are are other ways to avoid this problem, but adding zero and infinity seemed like the most logical to me

Reply:
`.infinity` would be useful if we need to check if a server is running a version less than a set version - the opposite of the example code I posted above.